### ISSUE TITLE
PHP7 – problems with "array" return type declaration

### ISSUE THREAD

**Issue by glen-84** [squizlabs/PHP_CodeSniffer#737]
```php
<?php

declare(strict_types = 1);

namespace X;

class Test
{
    public function test() : array
    {
        return [];
    }
}
```

Gives:

    Notice: Undefined index:  in /.../vendor/squizlabs/php_codesniffer/CodeSniffer/Standards/Squiz/Sniffs/Arrays/ArrayDeclarationSniff.php on line 82

    Notice: Undefined index:  in /.../vendor/squizlabs/php_codesniffer/CodeSniffer/Standards/Squiz/Sniffs/Arrays/ArrayDeclarationSniff.php on line 126

    Notice: Undefined index:  in /.../vendor/squizlabs/php_codesniffer/CodeSniffer/Standards/Squiz/Sniffs/Arrays/ArrayDeclarationSniff.php on line 126

    FILE: /.../app/core/src/Test.php
    ----------------------------------------------------------------------
    FOUND 2 ERRORS AFFECTING 1 LINE
    ----------------------------------------------------------------------
     9 | ERROR | [x] Short array syntax must be used to define arrays
       |       |     (Generic.Arrays.DisallowLongArraySyntax.Found)
     9 | ERROR | [x] There must be no space between the "array" keyword and
       |       |     the opening parenthesis
       |       |     (Squiz.Arrays.ArrayDeclaration.SpaceAfterKeyword)
    ----------------------------------------------------------------------
    PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
    ----------------------------------------------------------------------

    Time: 391ms; Memory: 4Mb

Versions:

* PHP 7.1.0-dev (cli) (built: Sep 29 2015 19:14:18) (DEBUG)
* squizlabs/php_codesniffer v2.3.4

**Comment by aik099** [comment]
That's because PHP 7 tokenizer might not be supported yet.

**Comment by gsherwood** [comment]
Again, PHP didn't assign a special token to return types, so the array return type was tokenized as T_ARRAY. I now detect return types and tokenize them a new T_RETURN_TYPE token instead.

Thanks for reporting.

**Comment by aik099** [comment]
@gsherwood , would this T_RETURN_TYPE also be used when other PHP 7 return types are used in function declaration, e.g. int, float, etc?

**Comment by gsherwood** [comment]
T_RETURN_TYPE should work for all return types.
